halo/hooks/object.cpp

```diff
@@ -146,116 +147,119 @@ extern "C" { // Demangle naming for ASM.
     type ## _set_last_update_time = def.set_last_update_time; \
     // End of Macro
 
-static ObjectBehaviorDefinition* backup[12];
+static ObjectBehaviorDefinition* vanilla_def_pointers_backup[POSITIVE_OBJECT_TYPES];
+const size_t OBJECT_DEF_PTR_ARRAY_ABS_SIZE = sizeof(uintptr_t)*POSITIVE_OBJECT_TYPES;
 
 void init_object_hooks(){
     auto game_defs = object_behavior_defs();
 
     // Copy data to our new defs.
 
-    const size_t s = sizeof(ObjectBehaviorDefinition);
+    const size_t S = sizeof(ObjectBehaviorDefinition);
```

Comment:
Stylistic nitpick:
1. Single character names are very hard to parse. Even a name like `obd_size` would be much better here.
2. While this value is technically a constant, you're defining it inline partway through a function, so you don't need to use CAPITAL_SNAKE_CASE, or `const`.